src/api.v2/model/UserAccess.js

```diff
@@ -27,6 +28,38 @@ class UserAccess extends BasicModel {
     super(sql, tableNames.USER_ACCESS, selectableProps);
   }
 
+  // Get all users and their access for this experiment
+  async getExperimentUsers(experimentId) {
+    const experimentsAccess = await this.find({ experiment_id: experimentId });
+
+    if (_.isEmpty(experimentsAccess)) {
+      throw new NotFoundError('Experiment not found');
+    }
+
+    // Remove admin from user list
+    const filteredAccess = experimentsAccess.filter(
+      ({ accessRole }) => accessRole !== AccessRole.ADMIN,
+    );
+
+    const requests = filteredAccess.map(
+      async (entry) => getAwsUserAttributesByEmail(entry.userId),
+    );
+
+    const cognitoUserData = await Promise.all(requests);
+
+    const experimentUsers = cognitoUserData.map((userInfo, idx) => {
+      const email = userInfo.find((attr) => attr.Name === 'email').Value;
+      const name = userInfo.find((attr) => attr.Name === 'name').Value;
+      const { accessRole } = filteredAccess[idx];
+
+      return {
+        name, email, role: accessRole,
+      };
+    });
```

Comment:
What do you think of perhaps just making the model return the sql response.

Then make the controller call a function that does all this part that interacts with aws cognito.

We could store this cognito function under a `cognito` folder inside `helpers` (that I think will be renamed to services probably)

Reply:
That way model remains a sql-only thing

Reply:
This depends on how we want to separate the concerns. From what I read, in MVC frameworks, people usually advocate for a ["thin controller, fat model"](https://stackoverflow.com/questions/14044681/fat-models-and-skinny-controllers-sounds-like-creating-god-models). Arguably, the controller's work in an MVC framework does not only clean input and receive data from a model, but also create the view, which our controller is not handling and make our controller feels very "thin".

I feel that anything business-logic or data-related should be placed in the model, so that we have only one place where data is calculated and processed. Doing the Cognito fetches in the controller would spread data processing in two layers.

Reply:
But this depends on how we want to structure the API.

Reply:
discussed over a call